File: swap/swap.js
function swap(a, b) {
    let tmp = a.value;
    a.value = b.value;
    b.value = tmp;
}

// Temporary variable swap
let a = { value: 5 };
let b = { value: 10 };
console.log("Using temporary variable swap:");
console.log(`Before swap: a = ${a.value}, b = ${b.value}`);
swap(a, b);
console.log(`After  swap: a = ${a.value}, b = ${b.value}`);
console.log("========================");

// XOR swap
let x = 15;
let y = 25;
console.log("Using XOR swap:");
console.log(`Before XOR swap: x = ${x}, y = ${y}`);
x = x ^ y;
y = x ^ y;
x = x ^ y;
console.log(`After  XOR swap: x = ${x}, y = ${y}`);
console.log("========================");

// Addition/subtraction swap
let m = 30;
let n = 45;
console.log("Using addition and subtraction swap:");
console.log(`Before add/sub swap: m = ${m}, n = ${n}`);
n = m - n;
m = m - n;
n = n + m;
console.log(`After  add/sub swap: m = ${m}, n = ${n}`);
console.log("========================");

File: swap/swap.c
#include <stdio.h>

void swap(int *a, int *b) {
    int tmp = *a;
    *a = *b;
    *b = tmp;
}

int main(int argc, char const *argv[]) {
    // Temporary variable & pointer swap example
    int a = 5;
    int b = 10;
    puts("Using temporary variable swap:");
    printf("Before swap: a = %d, b = %d\n", a, b);
    swap(&a, &b);
    printf("After  swap: a = %d, b = %d\n", a, b);
    puts("========================");

    // XOR swap example
    int x = 15;
    int y = 25;
    puts("Using XOR swap:");
    printf("Before XOR swap: x = %d, y = %d\n", x, y);
    x = x ^ y;  // x ^= y;
    y = x ^ y;  // y ^= x;
    x = x ^ y;  // x ^= y;
    // x ^= y ^= x ^= y;
    printf("After  XOR swap: x = %d, y = %d\n", x, y);
    puts("========================");

    // Addition and subtraction swap example
    int m = 30;
    int n = 45;
    puts("Using addition and subtraction swap:");
    printf("Before add/sub swap: m = %d, n = %d\n", m, n);
    n = m - n;
    m = m - n;  // m -= n;
    n = n + m;  // n += m;
    // n += m -= n = m - n;
    printf("After  add/sub swap: m = %d, n = %d\n", m, n);
    puts("========================");
    return 0;
}
